alacritty/src/renderer/rect.rs

```diff
@@ -0,0 +1,259 @@
+use std::mem::size_of;
+use std::ptr;
+
+use alacritty_terminal::term::SizeInfo;
+
+use crate::gl;
+use crate::gl::types::*;
+use crate::renderer::rects::RenderRect;
+use crate::renderer::{create_program, create_shader, Error, ShaderCreationError};
+
+/// Maxmimum number of rect vertices per batch. Limited by 16-bit integer indices.
+const MAX_VERTICES: usize = u16::max_value() as usize + 1;
```

Comment:
Can't we basically compute it in a way that it's `mod 4 == 0`, since rust allows constepxr stuff?

Reply:
+1 here is not for divisibility by four. It's there because max value for u16 is 65535, but max number of supported vertices is 65536 (65536th vertex indexed in base-0 will be 65535).